compiler/src/org.graalvm.compiler.truffle.runtime/src/org/graalvm/compiler/truffle/runtime/debug/disassembler/ObjdumpDisassembler.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.graalvm.compiler.truffle.runtime.debug.disassembler;
+
+import jdk.vm.ci.code.BytecodePosition;
+import jdk.vm.ci.code.site.Call;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Disassembles machine code by running it through objdump, which needs to be available on the path. Standard versions
+ * included in at least recent Linux distributions and macOS can be used.
+ */
+public class ObjdumpDisassembler implements Disassembler {
+
+    private static final Pattern ADDRESS_PATTERN = Pattern.compile("^([0-9a-f]+):\\s");
+    private static final Pattern RELATIVE_CODE_ADDRESS_PATTERN = Pattern.compile("<\\.text\\+0x([0-9a-f]+)>");
+
+    public String disassemble(MachineCodeAccessor machineCode) throws IOException {
+        final Process process = new ProcessBuilder()
+                .command("objdump", "--no-show-raw-insn", "-d", "/dev/stdin")
```

Comment:
In works in practice on macOS and Linux I think. I do know that then leaves Windows as an open problem.

But I will add checks and error messages.

Reply:
In works in practice on macOS and Linux I think - that option is common to both Gnu's implementation and Darwin's. I do know that then leaves Windows as an open problem.

But I will add checks and error messages.

Reply:
It would be great to add an engine option for additional disassembler options. Or at least an option to pass the equivalent of hsdis `-XX:PrintAssemblyOptions=intel` ('-x86-asm-syntax=intel' for objdump).